libminifi/src/core/ProcessSession.cpp

```diff
@@ -740,7 +740,7 @@ void ProcessSession::restore(const std::string &key, const std::shared_ptr<core:
   flow->clearStashClaim(key);
 }
 
-ProcessSession::RouteResult ProcessSession::routeFlowFile(const std::shared_ptr<FlowFile> &record) {
+ProcessSession::RouteResult ProcessSession::routeFlowFile(const std::shared_ptr<FlowFile> &record, std::unordered_map<std::string, TransferMetrics>& transfers) {
```

Comment:
I think it would be slightly nicer to make the second parameter a `std::function<void(const FlowFile&)> record_callback` and move the definition of the lambda to `commit()`.

Reply:
Updated in 5031ed6685c20f0d79617c32943b70bda2755eb0